src/compiler/checker.ts

```diff
@@ -8680,11 +8688,15 @@ namespace ts {
         }
 
         // Return the inferred type for a variable, parameter, or property declaration
-        function getTypeForVariableLikeDeclaration(declaration: ParameterDeclaration | PropertyDeclaration | PropertySignature | VariableDeclaration | BindingElement | JSDocPropertyLikeTag, includeOptionality: boolean): Type | undefined {
+        function getTypeForVariableLikeDeclaration(
+            declaration: ParameterDeclaration | PropertyDeclaration | PropertySignature | VariableDeclaration | BindingElement | JSDocPropertyLikeTag,
+            includeOptionality: boolean, // >> TODO: change this to use a flag in check mode; figure out how this interacts with caching

```

Comment:
If we decide to move forward with this PR, I'm going to replace the use of `includeOptionality` parameter by a new check mode flag.